salt/utils/extend.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+'''
+SaltStack Extend
+~~~~~~~~~~~~~~~~
+
+A templating tool for extending SaltStack.
+
+Takes a template directory and merges it into a SaltStack source code
+directory. This tool uses Jinja2 for templating.
+
+This tool is accessed using `salt-extend`
+
+    :codeauthor: :email:`Anthony Shaw <anthonyshaw@apache.org>`
+'''
+from __future__ import absolute_import
+from __future__ import print_function
+import yaml
+from datetime import date
+import tempfile
+import os
+import shutil
+from jinja2 import Template
+
+# zip compat for PY2/3
+from salt.ext.six.moves import zip
+from salt.utils import fopen
+from salt.utils.odict import OrderedDict
+
+import salt.version
+
+import logging
+log = logging.getLogger(__name__)
+
+try:
+    import click
+    HAS_CLICK = True
+except ImportError as ie:
+    HAS_CLICK = False
+
+TEMPLATE_FILE_NAME = 'template.yml'
+
+
+def _get_template(path, option_key):
+    '''
+    Get the contents of a template file and provide it as a module type
+
+    :param path: path to the template.yml file
+    :type  path: ``str``
+
+    :param option_key: The unique key of this template
+    :type  option_key: ``str``
+
+    :returns: Details about the template
+    :rtype: ``tuple``
+    '''
+    with fopen(path, "r") as template_f:
```

Comment:
@cachedout I dropped the namespace on the import https://github.com/saltstack/salt/pull/34921/files#diff-585fafce47a7c44a935dd239a8b93ff0R26